hollow/src/main/java/com/netflix/hollow/core/index/HollowHashIndex.java

```diff
@@ -146,10 +145,10 @@ private boolean matchIsEqual(FixedLengthElementArray matchHashTable, long hashBu
             int hashOrdinal = (int)matchHashTable.getElementValue(hashBucketBit + hashState.getOffsetPerTraverserField()[field.getBaseIteratorFieldIdx()], hashState.getBitsPerTraverserField()[field.getBaseIteratorFieldIdx()]) - 1;
 
             HollowTypeReadState readState = field.getBaseDataAccess();
-            int fieldPath[] = field.getSchemaFieldPositionPath();
+            int[] fieldPath = field.getSchemaFieldPositionPath();
 
             if(fieldPath.length == 0) {
-                if(hashOrdinal != ((Integer)query[i]).intValue())
+                if(hashOrdinal != (Integer) query[i])
```

Comment:
I guess we'd have to do that... 
BTW, this refactoring is relevant to Java 5.0 and higher. But hopefully it's ok.
This rule had been taken from [here](https://github.com/JetBrains/intellij-community/blob/master/plugins/InspectionGadgets/src/inspectionDescriptions/UnnecessaryUnboxing.html)

Reply:
Yup, definitely agree with the refactoring in general, thanks for that!

Reply:
To provide a bit of additional context: I only ask for the comparison change here because we've been bitten but this in the past, we had something like this but a refactoring changed the primitive to an Integer and equality checks started failing ([for values above 127](https://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html#valueOf%28int%29), so unit tests didn't catch them).

Reply:
I see now, it's for the sake of the future :)
It might look like that in this case:
`if (query[i].equals(hashOrdinal))
                    return false;`

Reply:
Sure, that works